apps/prairielearn/src/lib/db-types.ts

```diff
@@ -969,7 +969,7 @@ export const AssessmentAccessRuleSchema = z.object({
   end_date: DateFromISOString.nullable(),
   exam_uuid: z.string().nullable(),
   id: IdSchema,
-  mode: ModeSchema,
+  mode: ModeSchema.nullable(),
   number: z.number(),
```

Comment:
Technically number is also nullable in the DB (though I don't think it's typically null).